packages/wallet-sdk/src/util/provider.ts

```diff
@@ -61,10 +61,13 @@ export function getCoinbaseInjectedProvider({
     }
   }
 
-  const ethereum = window.ethereum ?? window.top?.ethereum;
```

Comment:
how about creating a dedicated function or something to do just this and catch error?
the if statement below doesn’t throw error anyway.